paddle/fluid/memory/allocation/legacy_allocator.cc

```diff
@@ -337,13 +339,14 @@ namespace allocation {
 
 Allocation *LegacyAllocator::AllocateImpl(size_t size, Allocator::Attr attr) {
   void *ptr = boost::apply_visitor(legacy::AllocVisitor(size), place_);
+  platform::RecordMemEvent record_mem_event(true, size, place_);
   return new Allocation(ptr, size, place_);
 }
 
 void LegacyAllocator::Free(Allocation *allocation) {
-  boost::apply_visitor(
-      legacy::FreeVisitor(allocation->ptr(), allocation->size()),
-      allocation->place());
+  boost::apply_visitor(legacy::FreeVisitor(allocation->ptr()),
+                       allocation->place());
+  platform::RecordMemEvent record_mem_event(false, allocation->size(), place_);
```

Comment:
Add comment to describe the meaning of `false`, e.g. `(false /*is_alloc*/, ...`.